STYLE_GUIDE.md

```diff
@@ -0,0 +1,185 @@
+  <h1>Contact us</h1>
+  <p>Use this form to send us a message.</p>
+  <label htmlFor="email">Email address:</label>
+  <input id="email" type="email" />
+  <button>Submit</button>
+</form>
+```
+
+## CSS
+
+### Class names
+
+This project uses the [BEM naming convention](http://getbem.com/), to avoid conflicts, reduce rule specificity, and make classes easier to read and understand. [Reducing rule specificity](https://css-tricks.com/strategies-keeping-css-specificity-low/) is particularly important for improving CSS maintainability, so we try to avoid nesting CSS selectors as much as possible.
+
+We use the `pipeline-` prefix for most HTML/CSS classes, to reduce the risk of clashes in the global namespace when this project is imported into other applications. It's important never 
+
+<!-- @TODO Explain usage of the 'kedro' class, Kedro-UI, and webfonts -->
+
+### Variables
+
+We use Sass variables to declare colours and reused values wherever useful. It's not necessary to declare every CSS unit value as a variable, but if a value is used in multiple places (for related reasons), then it makes sense to make it a variable - either locally in the component, or application-wide in `src/styles/_variables.scss`. Using a named variable instead of a number can clarify your intent, and help you avoid having to update a value in multiple places later on.
+
+Avoid:
+```scss
+color: #001521;
+color: rgba(#001521, 0.5);
+color: rgba(0, 0, 0, 0.5);
+```
+
+Prefer:
+```scss
+color: $color-bg-dark-1;
+color: rgba($color-bg-dark-1, 0.5);
+color: rgba(black, 0.5);
+```
+
+### Theme colours
+
+Kedro-Viz has both light and dark themes, and every component must be styled appropriately for each theme. The themes are set using the same theme classes that Kedro-UI uses - `kui-theme--light` and `kui-theme--dark` - applied on the top-level container element. In order to simplify code and avoid increasing CSS specificity, we use CSS Custom Properties to handle theme colours, and we recommend that you use these where possible.
+
+Avoid:
+```scss
+.pipeline-title {
+  .kui-theme--light & {
+    background: $color-bg-light-1;
+  }
+
+  .kui-theme--dark & {
+    background: $color-bg-dark-1;
+  }
+}
+```
+
+Prefer:
+```scss
+.kui-theme--light {
+  --color-bg-1: $color-bg-light-1;
+}
+
+.kui-theme--dark {
+  --color-bg-1: $color-bg-dark-1;
+}
+
+.pipeline-title {
+  background: var(--color-bg-1);
+}
+```
+
+<!-- @TODO -->
+
+### Units (em, %, px, etc)
+
+<!-- @TODO -->
+
+## JavaScript
+
+### Imports
+
+Although it will work in regular development (because this project uses React-Scripts), avoid importing non-standard file-types like `.scss` and `.svg` directly into JavaScript files. This is because these won't work without specific webpack loaders, so when Kedro-Viz is imported into other projects, it will break. The `lib-test` testing suite exists partly for this reason, to check that Kedro-Viz still works when imported as a componnent library into a fairly standard JS app.
```

Comment:
```suggestion
Although it works in regular development (because this project uses React-Scripts), you should avoid importing non-standard file-types like `.scss` and `.svg` directly into JavaScript files. This is because these won't work without specific webpack loaders, and breaks when Kedro-Viz is imported into other projects. The `lib-test` testing suite exists partly for this reason, to check that Kedro-Viz still works when imported as a component library into a fairly standard JS app.
```